### ISSUE TITLE
"Failed to find playbook" error when importing a playbook from a collection

### ISSUE THREAD

**Issue by acidroper** [ansible/ansible-lint#4362]
##### Summary

When using `import_playbook` with a playbook from a collection, `ansible-lint` reports the following error:
```
ERROR    Failed to find my_namespace.my_collection.playbook playbook.
```

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 24.9.2 using ansible-core:2.16.11 ansible-compat:24.9.1 ruamel-yaml:0.18.6 ruamel-yaml-clib:0.2.8
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

Write a playbook that imports a playbook from an installed collection:

```yaml
---
- name: Import a playbook
  ansible.builtin.import_playbook: my_namespace.my_collection.playbook
```

Run `ansible-lint import_playbook.yml`

##### Desired Behavior

No errors are expected.

##### Actual Behavior

```console (paste below)
$ ansible-lint import_playbook.yml
ERROR    Failed to find my_namespace.my_collection.playbook playbook.

Passed: 0 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'production'.

$ ls -l ~/.ansible/collections/ansible_collections/my_namespace/my_collection/playbooks/playbook.yml
-rw-r--r-- 1 user user 187 окт  9 10:42 /home/user/.ansible/collections/ansible_collections/my_namespace/my_collection/playbooks/playbook.yml
```

**Comment by acidroper** [comment]
Thanks for the fix. However, there is still an issue: if a playbook is located in a nested directory, ansible-lint still reports the error.

**Comment by cavcrosby** [comment]
@acidroper Could you provide me an example reproducing the issue?

**Comment by acidroper** [comment]
@cavcrosby  Sure.

```yaml
---
- name: Import a nested playbook
  ansible.builtin.import_playbook: my_namespace.my_collection.test.playbook
```

```console
$ ansible-lint import_playbook.yml
ERROR    Failed to find my_namespace.my_collection.test.playbook playbook.

Passed: 0 failure(s), 0 warning(s) on 1 files. Last profile that met the validation criteria was 'production'.
```

```console
$ ls -l ~/.ansible/collections/ansible_collections/my_namespace/my_collection/playbooks/test/playbook.yml                                                                                                                 
-rw-r--r-- 1 user user 233 ноя 15 11:16 /home/user/.ansible/collections/ansible_collections/my_namespace/my_collection/playbooks/test/playbook.yml
```
